cmd/virt-handler/virt-handler.go

```diff
@@ -73,7 +74,13 @@ func main() {
 	broadcaster.StartRecordingToSink(&kubecorev1.EventSinkImpl{Interface: coreClient.Events(api.NamespaceDefault)})
 	recorder := broadcaster.NewRecorder(kubev1.EventSource{Component: "virt-handler", Host: *host})
 
-	domainManager, err := virtwrap.NewLibvirtDomainManager(domainConn, recorder)
+	var domainManager virtwrap.DomainManager
+	switch *driver {
+	case "ovirt":
+		domainManager, err = virtwrap.NewOvirtDomainManager(domainConn, recorder)
+	default:
+		domainManager, err = virtwrap.NewLibvirtDomainManager(domainConn, recorder)
+	}
```

Comment:
well it's certainly not intended to stay like that for too long. The intent was to incrementally reduce the scope of vdsm interactions needed while kubevirt improves.
So I wouldn't suggest to start building an elaborated interface and hook points for things oVirt people are actively trying to remove.